src/snapshot.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const { Session } = require('inspector')
+const { openSync, closeSync, writeSync } = require('fs')
+const { ensurePromiseCallback, destinationFile } = require('./utils')
+
+module.exports = function generateHeapSnapshot(cb) {
+  // Prepare the context
+  const [callback, promise] = ensurePromiseCallback(cb)
+  const destination = destinationFile('heapsnapshot')
+  const session = new Session()
+  let fd
+  let writeError
+
+  // Open the destination file
+  try {
+    fd = openSync(destination, 'w')
+  } catch (e) {
+    return callback(e)
+  }
+
+  // Start the session
+  session.connect()
```

Comment:
can session emit 'error'? In case, we should call the calback

Reply:
Apparently no, it cannot.